File: capy/src/e2e-tests/java/dev/capylang/test/SameNameObjectOrientedFpInteropTest.java
package dev.capylang.test;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class SameNameObjectOrientedFpInteropTest {
    @Test
    void objectOrientedCodeCanUseFpSiblingModuleWithoutImportsWhenFileNamesMatch() {
        var interactor = new SharedInteractor();

        assertThat(interactor.invoke_fp_function("Capy")).isEqualTo("dog:Capy");
        assertThat(interactor.create_fp_data("Bara")).isInstanceOf(SharedInterop.SharedDog.class);
        assertThat(((SharedInterop.SharedDog) interactor.create_fp_data("Bara")).name()).isEqualTo("Bara");
        assertThat(interactor.match_fp_type("Mochi")).isEqualTo("dog:Mochi");
    }
}

File: capy/src/e2e-coo/js/dev/capylang/test/SameNameObjectOrientedFpInterop.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('object-oriented code can use FP sibling module without imports when file names match', () => {
    const { SharedInteractor } = generatedModule('dev/capylang/test/SharedInteractor.js');
    const shared = generatedModule('dev/capylang/test/SharedInterop.js');
    const interactor = new SharedInteractor();

    assert.equal(interactor.invoke_fp_function('Capy'), 'dog:Capy');
    const dog = interactor.create_fp_data('Bara');
    assert.equal(dog instanceof shared.SharedDog, true);
    assert.equal(dog.name, 'Bara');
    assert.equal(interactor.match_fp_type('Mochi'), 'dog:Mochi');
});
